system/HTTP/UserAgent.php

```diff
@@ -130,14 +130,11 @@ class UserAgent
 	 *
 	 * Sets the User Agent and runs the compilation routine
 	 *
-	 * @param null $config
+	 * @param null|\Config\UserAgents $config
 	 */
-	public function __construct($config = null)
+	public function __construct(?UserAgents $config = null)
```

Comment:
I don't think the `?` is needed here since the default value is null.

Reply:
I know that `= null` here already implies nullability of the type but I have read somewhere that it is perfectly valid.

Reply:
This is from the original [RFC](https://wiki.php.net/rfc/nullable_types):

![image](https://user-images.githubusercontent.com/51850998/82832899-18488680-9eef-11ea-8afd-5d9463b82e9b.png)

But if you're against it, I can still remove that. :)

Reply:
I would prefer to use the explicit `?` in case the default changes. On the other hand its redundant.

Reply:
Basing on other code within the codebase, the `?` is indeed not used. Changing..